added dal, business-logic

Gashad0kuro opened the pull request:
added a single table "wines" + entity class and a controller to test functionality of repository and business-logic

FrediKats commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/TestWineController.java:
```diff
@@ -0,0 +1,42 @@
+public class TestWineController {
+    private final WineService wineService;
+
+    @GetMapping("/all")
```
Не по REST-у немного)
Емнип, бест практис роутинга - это дефолтный круд:
GET /api/my-entity-name - get all
GET /api/my-entity-name/{id} - get by id
POST /api/my-entity-name - add new

FrediKats commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/TestWineController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @PostMapping("/add")
+    public void addWine(@RequestParam String brand, @RequestParam String color, @RequestParam String grape,
+            @RequestParam String name, @RequestParam Integer price, @RequestParam String region,
+            @RequestParam Float strength, @RequestParam String sugar, @RequestParam Float volume) {
```
Дто и через тело запроса? У get'а не лимит есть длинны, когда-то выстрелит)

FrediKats commented on src/main/java/com/wine/to/up/winestyle/parser/service/domain/entity/Wine.java:
```diff
@@ -0,0 +1,47 @@
+        this.sugar = sugar;
+        this.volume = volume;
+    }
+
+    
+
+
```
meh

FrediKats commented on src/main/java/com/wine/to/up/winestyle/parser/service/services/WineService.java:
```diff
@@ -0,0 +1,33 @@
+package com.wine.to.up.winestyle.parser.service.services;
+
+import com.wine.to.up.winestyle.parser.service.domain.entity.Wine;
+
+import java.util.List;
+
+import com.wine.to.up.winestyle.parser.service.repository.WineRepository;
+
+import lombok.RequiredArgsConstructor;
```
А по какому принципу тут отступы делаются? Во всех файлах выше еще просматривалась логика,.

shinehlina replied:
Согласна)

FrediKats reviewed (changes requested):
Руты бы переделать под рест. Кажется, на лекциях еще не рассказывали, тогда стоит собрать эту инфу и возможно зашарить между лидами.

baby-benz committed:
embed parser application codebase in winestyle-parser-service

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/MessageController.java:
```diff
@@ -2,7 +2,7 @@
 import com.wine.to.up.winestyle.parser.service.repository.MessageRepository;
 import lombok.RequiredArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
+//import lombok.extern.slf4j.Slf4j;
```
?

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/TestWineController.java:
```diff
@@ -0,0 +1,42 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import com.wine.to.up.winestyle.parser.service.domain.entity.Wine;
+//import com.wine.to.up.winestyle.parser.service.repository.WineRepository;
```
?

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/services/WineService.java:
```diff
@@ -0,0 +1,33 @@
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+
```
Тут какая-то вообще с отступами интересная история

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/services/WineService.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    // add new wine
+    public void addNew(String brand, String color, String grape, String name, Integer price, String region,
```
Это реально выглядит длинновато

NShmendyuk committed:
added cron expression in application properties, check service business on scheduler

baby-benz committed:
large scale project restructuring
change getElementBySmth to selectFirst for performance increase purposes
implement builder pattern by adding additional @Builder and constructor annotations
tell JPA Hibernate to autogenerate table by adding spring.jpa.hibernate.ddl-auto=update to .properties file

baby-benz committed:
Merge remote-tracking branch 'origin/feature/scheduler-upgrade' into dev

# Conflicts:
#	src/main/java/com/wine/to/up/winestyle/parser/service/service/ParserService.java

baby-benz committed:
minor scheduler and entity properties fixes

baby-benz committed:
fix initial condition in parseByPage loop

Commit:
Javadoc added
